packages/relayer/processor/process_message.go

```diff
@@ -461,3 +452,45 @@ func (p *Processor) saveMessageStatusChangedEvent(
 
 	return nil
 }
+
+func (p *Processor) setGasTipOrPrice(ctx context.Context, auth *bind.TransactOpts) error {
+	gasTipCap, err := p.destEthClient.SuggestGasTipCap(ctx)
+	if err != nil {
+		if IsMaxPriorityFeePerGasNotFoundError(err) {
+			auth.GasTipCap = FallbackGasTipCap
+		} else {
+			gasPrice, err := p.destEthClient.SuggestGasPrice(context.Background())
+			if err != nil {
+				return errors.Wrap(err, "p.destBridge.SuggestGasPrice")
+			}
+			auth.GasPrice = gasPrice
+		}
+	} else {
+		auth.GasTipCap = gasTipCap
+
+		return nil
+	}
+
+	return nil
+}
+
+func (p *Processor) getCost(ctx context.Context, auth *bind.TransactOpts) (*big.Int, error) {
+	if auth.GasTipCap != nil {
+		bn, err := p.destEthClient.BlockNumber(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		blk, _ := p.destEthClient.BlockByNumber(context.Background(), new(big.Int).SetUint64(bn))
```

Comment:
```suggestion
		blk, _ := p.destEthClient.BlockByNumber(ctx, new(big.Int).SetUint64(bn))
```

can't understand why use `context.Background()` here